src/read/xcoff/relocation.rs

```diff
@@ -34,8 +37,34 @@ where
     type Item = (u64, Relocation);
 
     fn next(&mut self) -> Option<Self::Item> {
-        // TODO: return the relocations in the section.
-        None
+        self.relocations.next().map(|relocation| {
+            let encoding = RelocationEncoding::Generic;
+            let (kind, addend) = match relocation.r_rtype() {
+                xcoff::R_POS
+                | xcoff::R_RL
+                | xcoff::R_RLA
+                | xcoff::R_BA
+                | xcoff::R_RBA
+                | xcoff::R_TLS => (RelocationKind::Absolute, 0),
```

Comment:
Can you explain your reasoning for `R_TLS` being absolute? I'm not sure about it. It might be okay, and there may not be a better alternative.

Reply:
It seems treated as absolute in [llvm](https://github.com/llvm/llvm-project/blob/902ba8b0c9b013043aa04dc548be3ec907ef5571/llvm/lib/MC/XCOFFObjectWriter.cpp#L567)

```
  if (Type == XCOFF::RelocationType::R_POS ||
      Type == XCOFF::RelocationType::R_TLS)
    // The FixedValue should be symbol's virtual address in this object file
    // plus any constant value that we might get.
    FixedValue = getVirtualAddress(SymA, SymASec) + Target.getConstant();
```